### PR TITLE
New rules for Lambda runtime EOL

### PR DESCRIPTION
**Pull request by kddejong**
*Issue #, if available:*

*Description of changes:*
- New rule E2531 to check if a lambda runtime is deprecated
- New rule W2531 to check if a lambda runtime is EOL

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/cfnlint/rules/resources/lmbd/DeprecatedRuntime.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+  HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+  OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+  SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+"""
+from datetime import datetime
+from cfnlint import CloudFormationLintRule
+
+
+class DeprecatedRuntime(CloudFormationLintRule):
```
Comment (fatbasstard) [inline src/cfnlint/rules/resources/lmbd/DeprecatedRuntime.py]: Doesn't this run as a standalone rule? Or is it filtered out based on the fact that there is no Meta data specified?

File: src/cfnlint/rules/resources/lmbd/DeprecatedRuntime.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    """Check if EOL Lambda Function Runtimes are used"""
+
+    def __init__(self):
+        """Init"""
+        super(DeprecatedRuntime, self).__init__()
+        self.resource_property_types.append('AWS::Lambda::Function')
+
+    current_date = datetime.today()
+
+    deprecated_runtimes = {
```
Comment (fatbasstard) [inline src/cfnlint/rules/resources/lmbd/DeprecatedRuntime.py]: Maybe move this to a JSON file (or make it part of the Spec files, but that might be interesting for later)?

Comment (kddejong) [reply src/cfnlint/rules/resources/lmbd/DeprecatedRuntime.py]: yea, I like that idea but any thoughts about how we would fold this into the spec file?

Comment (fatbasstard) [reply src/cfnlint/rules/resources/lmbd/DeprecatedRuntime.py]: For now just a standalone (like `OnlyOne.json`), to keep it consistent?

File: src/cfnlint/rules/resources/lmbd/DeprecatedRuntime.py (hunk 1 above)
Comment (kddejong) [reply src/cfnlint/rules/resources/lmbd/DeprecatedRuntime.py]: It is filtered out so it doesn't run.

File: src/cfnlint/rules/resources/lmbd/DeprecatedRuntime.py (hunk 2 above)
Comment (kddejong) [reply src/cfnlint/rules/resources/lmbd/DeprecatedRuntime.py]: Moved.

**Pull request merged**